Dockerfile

```diff
@@ -1,17 +1,12 @@
-FROM golang:alpine as build
-LABEL org.opencontainers.image.source = "https://github.com/containerish/OpenRegistry"
-
+FROM golang:1.17 as build
+ENV CGO_ENABLED=0
 WORKDIR /root/openregistry
-
-COPY Makefile .
-RUN apk add gcc make git curl ca-certificates
-
 COPY . .
-RUN go mod download && CGO_ENABLED=0 go build -o openregistry -ldflags="-w -s" main.go
+RUN go build -o openregistry -ldflags="-w -s" -trimpath main.go
 
 FROM alpine:latest
-
-COPY --from=build /root/openregistry/openregistry .
+LABEL org.opencontainers.image.source = "https://github.com/containerish/OpenRegistry"
+COPY --from=build /root/openregistry/openregistry /bin/openregistry
 COPY config.yaml .
 EXPOSE 5000
-CMD ["./openregistry"]
+ENTRYPOINT ["openregistry"]
```

Comment:
Also I don't see any specific benefit over `CMD`? I usually prefer `CMD` since the binary runs as PID 1 and makes it impossible to force the container to use a different start command? But please let me know `ENTRYPOINT` is suited better?

Reply:
> I usually prefer CMD since the binary runs as PID 1 and makes it impossible to force the container to use a different start command?

That's `entrypoint` that you're describing.

As I said in PR comment, `entrypoint` receive OS signals like `SIGSTOP` which can be utilized for graceful shutdown. This is best practice.